politeiad/cmd/legacypoliteia/proposal.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+
+	backend "github.com/decred/politeia/politeiad/backendv2"
+	"github.com/decred/politeia/politeiad/plugins/comments"
+	"github.com/decred/politeia/politeiad/plugins/pi"
+	"github.com/decred/politeia/politeiad/plugins/ticketvote"
+	"github.com/decred/politeia/politeiad/plugins/usermd"
+)
+
+// proposal contains the full contents of a tstore proposal.
+type proposal struct {
+	RecordMetadata backend.RecordMetadata
+
+	// Files includes the proposal index file and image attachments.
+	Files []backend.File
+
+	// The following fields are converted into backend files before being
+	// imported into tstore.
+	//
+	// The VoteMetadata will only exist for RFPs and RFP submissions.
+	ProposalMetadata pi.ProposalMetadata
+	VoteMetadata     *ticketvote.VoteMetadata
+
+	// The following fields are converted into backend metadata streams before
+	// being imported into tstore.
+	UserMetadata usermd.UserMetadata
+	StatusChange *usermd.StatusChangeMetadata
```

Comment:
This is an optional pointer, but there's no documentation on when it will be nil.